Bottom: 6d149a1cd688a8ebd98ea7ef7a131d6f396ce280
Top:    520f34257619c1820277617e4e2305e2bdd0fde6
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 14:07:09 -0500

Add buildbot PTXdist worker Docker image

---
diff --git a/ansible-playbook/group_vars/all/buildbot.yml b/ansible-playbook/group_vars/all/buildbot.yml
index d5a1a447..29d7c9f8 100644
--- a/ansible-playbook/group_vars/all/buildbot.yml
+++ b/ansible-playbook/group_vars/all/buildbot.yml
@@ -2,3 +2,4 @@
 buildbot_version: 1.1.1
 buildbot_worker_port: 9989
 buildbot_worker_ntel: worker-ntel
+buildbot_worker_ptxdist: worker-ptxdist
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 3fa1e99f..c5ebfa0d 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -19,3 +19,4 @@ buildsystem_database_build_password: '{{ lookup("password", inventory_dir + "/cr
 svn_username:                        '{{ lookup("file",     inventory_dir + "/personal_credentials/svn/username") }}'
 svn_password:                        '{{ lookup("file",     inventory_dir + "/personal_credentials/svn/password") }}'
 buildbot_worker_ntel_password:       '{{ lookup("password", inventory_dir + "/credentials/buildbot.novatech-llc.com/worker_ntel length=20") }}'
+buildbot_worker_ptxdist_password:    '{{ lookup("password", inventory_dir + "/credentials/buildbot.novatech-llc.com/worker_ntel length=20") }}'
diff --git a/roles/buildbot-worker-ptxdist/README.md b/roles/buildbot-worker-ptxdist/README.md
new file mode 100644
index 00000000..225dd44b
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/README.md
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+
+Role Variables
+--------------
+
+A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+
+Dependencies
+------------
+
+A list of other roles hosted on Galaxy should go here, plus any details in regards to parameters that may need to be set for other roles, or variables that are used from other roles.
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
new file mode 100644
index 00000000..9cdd8b6f
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -0,0 +1,23 @@
+---
+# defaults file for buildbot-worker-ptxdist
+
+# The hostname passed as an envirnment variable into the container
+container_addr_map:
+  buildbot:
+    ip_addr: '127.0.0.1'
+    hostname: buildbot-worker-ptxdist.example.com
+
+# Network names
+docker_network_frontend: 'frontend'
+
+buildbot_worker_ptxdist_data_path: '/buildbot'
+
+# the name of the image being duilt and used for the container
+buildbot_worker_ptxdist_image_repo: https://github.com/novatechweb/docker-buildbot-worker-ptxdist.git
+buildbot_worker_ptxdist_image_name: '{{ docker_registry_username }}/buildbot-worker-ptxdist'
+
+# the name of the container being started
+buildbot_worker_ptxdist_container_name: '{{ docker_name_prefix }}buildbot-worker-ptxdist'
+
+# the name of the data-volume used by the container
+buildbot_worker_ptxdist_dv_name: '{{ docker_name_prefix }}buildbot-worker-ptxdist_DV'
diff --git a/roles/buildbot-worker-ptxdist/handlers/main.yml b/roles/buildbot-worker-ptxdist/handlers/main.yml
new file mode 100644
index 00000000..78087ae9
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/handlers/main.yml
@@ -0,0 +1,2 @@
+---
+# handlers file for buildbot-worker-ptxdist
\ No newline at end of file
diff --git a/roles/buildbot-worker-ptxdist/meta/main.yml b/roles/buildbot-worker-ptxdist/meta/main.yml
new file mode 100644
index 00000000..72237997
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/meta/main.yml
@@ -0,0 +1,57 @@
+galaxy_info:
+  author: your name
+  description: your description
+  company: your company (optional)
+
+  # If the issue tracker for your role is not on github, uncomment the
+  # next line and provide a value
+  # issue_tracker_url: http://example.com/issue/tracker
+
+  # Some suggested licenses:
+  # - BSD (default)
+  # - MIT
+  # - GPLv2
+  # - GPLv3
+  # - Apache
+  # - CC-BY
+  license: license (GPLv2, CC-BY, etc)
+
+  min_ansible_version: 1.2
+
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  # Optionally specify the branch Galaxy will use when accessing the GitHub
+  # repo for this role. During role install, if no tags are available,
+  # Galaxy will use this branch. During import Galaxy will access files on
+  # this branch. If Travis integration is configured, only notifications for this
+  # branch will be accepted. Otherwise, in all cases, the repo's default branch
+  # (usually master) will be used.
+  #github_branch:
+
+  #
+  # platforms is a list of platforms, and each platform has a name and a list of versions.
+  #
+  # platforms:
+  # - name: Fedora
+  #   versions:
+  #   - all
+  #   - 25
+  # - name: SomePlatform
+  #   versions:
+  #   - all
+  #   - 1.0
+  #   - 7
+  #   - 99.99
+
+  galaxy_tags: []
+    # List tags for your role here, one per line. A tag is a keyword that describes
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
\ No newline at end of file
diff --git a/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh b/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh
new file mode 100755
index 00000000..8353ad9d
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+DIRS="/cache \
+      /cache/ccache \
+      /cache/downloads \
+      /cache/images \
+      /cache/ipkg-repository \
+      /cache/mirrors \
+      /cache/releases \
+      /cache/sstate"
+
+for d in $DIRS
+do
+  mkdir -p $d
+  chown -c -R buildbot:buildbot $d
+  chmod -v 777 $d
+done
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
new file mode 100644
index 00000000..9bc8c930
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -0,0 +1,30 @@
+---
+# tasks file for buildbot-worker-ptxdist
+
+# This will generate passwords for these accounts.
+- assert:
+    that:
+    - buildbot_worker_ptxdist_password is defined
+
+- name: Create buildbot data volume
+  docker_volume:
+    name: '{{ buildbot_dv_name }}'
+
+- name: Checkout image repo
+  git:
+    repo: '{{ buildbot_worker_ptxdist_image_repo }}'
+    version: master
+    dest: '{{ docker_projects_dir }}/buildbot-worker-ptxdist'
+
+- name: Create buildbot master image
+  docker_image:
+    name: '{{ buildbot_worker_ptxdist_image_name }}'
+    tag: '{{ docker_image_tag }}'
+    path: '{{ docker_projects_dir }}/buildbot-worker-ptxdist'
+
+- name: Start buildbot container
+  docker_container:
+    name: '{{ buildbot_worker_ptxdist_container_name }}'
+    image: '{{ buildbot_worker_ptxdist_image_name }}:{{ docker_image_tag }}'
+    volumes:
+      - '{{ buildbot_worker_ptxdist_dv_name }}:{{ buildbot_worker_ptxdist_data_path }}:z'
diff --git a/roles/buildbot-worker-ptxdist/tests/inventory b/roles/buildbot-worker-ptxdist/tests/inventory
new file mode 100644
index 00000000..878877b0
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/tests/inventory
@@ -0,0 +1,2 @@
+localhost
+
diff --git a/roles/buildbot-worker-ptxdist/tests/test.yml b/roles/buildbot-worker-ptxdist/tests/test.yml
new file mode 100644
index 00000000..2d00de44
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/tests/test.yml
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - buildbot-worker-ptxdist
\ No newline at end of file
diff --git a/roles/buildbot-worker-ptxdist/vars/main.yml b/roles/buildbot-worker-ptxdist/vars/main.yml
new file mode 100644
index 00000000..ae5fa61c
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for buildbot-worker-ptxdist
\ No newline at end of file
